packages/flutter/lib/src/material/expand_icon.dart

```diff
@@ -179,7 +185,7 @@ class _ExpandIconState extends State<ExpandIcon> with SingleTickerProviderStateM
         onPressed: widget.onPressed == null ? null : _handlePressed,
         icon: RotationTransition(
           turns: _iconTurns,
-          child: const Icon(Icons.expand_more),
+          child: widget.iconData ? const Icon(widget.iconData) : const Icon(Icons.expand_more),
```

Comment:
Looks like a logic issue here that is causing all kinds of check failures. Should probably be something like:

```suggestion
          child: const Icon(widget.iconData ?? Icons.expand_more),
```